devito/types/dimension.py

```diff
@@ -1060,6 +1064,36 @@ def __new__(cls, d, name):
     _pickle_kwargs = []
 
 
+class CustomDimension(BasicDimension):
+
+    """
+    Dimension defining an iteration space with custom (known) size.
+
+    Notes
+    -----
+    This could be extended in the future for more customization (e.g., min point,
+    max point, etc.).
+    """
+
+    is_Custom = True
+
+    def __init_finalize__(self, name, symbolic_size=None):
+        self._symbolic_size = symbolic_size
+
+    @property
+    def symbolic_size(self):
+        return self._symbolic_size
+
+    def _arg_defaults(self):
+        return {}
+
+    def _arg_values(self, *args, **kwargs):
```

Comment:
the BasicDimension  `_arg` don't work? I gues purely symbolic so no actual size.

Reply:
indeed. This is on purpose a no-op, like we do for the other Dimensions that are expected not to provide any arguments